distributed/profile.py

```diff
@@ -275,24 +275,42 @@ def traverse(state, start, stop, height):
     }
 
 
+_watch_running: set[int] = set()
+
+
+def wait_profiler() -> None:
+    """Wait until a moment when no instances of watch() are sampling the frames.
+    You must call this function whenever you would otherwise expect an object to be
+    immediately released after it's descoped.
+    """
+    while _watch_running:
+        sleep(0.0001)
```

Comment:
I'm not entirely sure how I feel about this. It is not thread safe so while this function is returning another profile thread might again sample the frames. At the same time, if a profile thread is actually running, this thing will burn CPU hard.
For this specific use case, that's likely not a problem but I'm wondering if a `sleep(0.1)` would not have an equal power without us putting this kind of instrumentation in.

Reply:
I agree, this is OK only as long as its usage is limited to unit tests